crypto/dh/dh_ameth.c

```diff
@@ -539,6 +540,26 @@ static int dh_pkey_export_to(const EVP_PKEY *from, void *to_keydata,
     return rv;
 }
 
+static int dh_pkey_import_from(const OSSL_PARAM params[], void *key)
+{
+    EVP_PKEY *pkey = key;
+    DH *dh = DH_new();
+
+    if (dh == NULL) {
+        ERR_raise(ERR_LIB_DH, ERR_R_MALLOC_FAILURE);
+        return 0;
+    }
+
+    if (!ffc_fromdata(dh_get0_params(dh), params)
+        || !dh_key_fromdata(dh, params)) {
+        DH_free(dh);
+        return 0;
+    }
+
+    EVP_PKEY_assign_DH(pkey, dh);
```

Comment:
probably should be checking the return code here.

Reply:
Ah, good point

Reply:
(although, looking through old code, you'll notice that we seldom do that...  still, will fix)

Reply:
Fixed in all legacy import functions